file/src/find.rs

```diff
@@ -0,0 +1,400 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+use std::collections::HashSet;
+use std::{env, fs};
+use std::os::unix::fs::{FileTypeExt, MetadataExt, PermissionsExt};
+use std::path::PathBuf;
+use std::process::ExitCode;
+use gettextrs::{bind_textdomain_codeset, textdomain};
+use plib::PROJECT_NAME;
+use walkdir::{DirEntry, WalkDir};
+
+#[derive(Debug)]
+enum Expr {
+    And(Vec<Expr>, Vec<Expr>),
+    Or(Vec<Expr>, Vec<Expr>),
+    Not(Vec<Expr>),
+    Name(String),
+    MTime(i64),
+    Path(String),
+    Type(char),
+    NoUser,
+    NoGroup,
+    XDev,
+    Prune,
+    Perm(u32),
+    Links(u64),
+    User(String),
+    Group(String),
+    Size(u64, bool),
+    Print,
+    Newer(PathBuf),
+}
+
+/// Parses a list of tokens representing search criteria, the result is stored in enum Expr
+/// 
+/// # Arguments
+///
+/// * `tokens` - A vector of string slices representing the tokens to parse.
+///
+/// # Returns
+///
+/// * A vector of `Expr` expressions parsed from the tokens.
+fn parse_expression(tokens: &mut Vec<&str>) -> Vec<Expr> {
+    let mut stack: Vec<Expr> = Vec::new();
+
+    while let Some(&token) = tokens.last() {
+        match token {
+            "-name" => {
+                tokens.pop();
+                if let Some(name) = tokens.pop() {
+                    stack.push(Expr::Name(name.to_string()));
+                }
+            }
```

Comment:
-name arg doesn't accept the name of a file (or even a part of the name). It accepts a pattern that is a bit more complex to match by: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_13